src/messages.rs

```diff
@@ -1324,6 +1325,92 @@ impl ApiKey {
+            ApiKey::DescribeDelegationTokenKey => VersionRange { min: 0, max: 3 },
+            ApiKey::DeleteGroupsKey => VersionRange { min: 0, max: 2 },
+            ApiKey::ElectLeadersKey => VersionRange { min: 0, max: 2 },
+            ApiKey::IncrementalAlterConfigsKey => VersionRange { min: 0, max: 1 },
+            ApiKey::AlterPartitionReassignmentsKey => VersionRange { min: 0, max: 0 },
+            ApiKey::ListPartitionReassignmentsKey => VersionRange { min: 0, max: 0 },
+            ApiKey::OffsetDeleteKey => VersionRange { min: 0, max: 0 },
+            ApiKey::DescribeClientQuotasKey => VersionRange { min: 0, max: 1 },
+            ApiKey::AlterClientQuotasKey => VersionRange { min: 0, max: 1 },
+            ApiKey::DescribeUserScramCredentialsKey => VersionRange { min: 0, max: 0 },
+            ApiKey::AlterUserScramCredentialsKey => VersionRange { min: 0, max: 0 },
+            ApiKey::VoteKey => VersionRange { min: 0, max: 0 },
+            ApiKey::BeginQuorumEpochKey => VersionRange { min: 0, max: 0 },
+            ApiKey::EndQuorumEpochKey => VersionRange { min: 0, max: 0 },
+            ApiKey::DescribeQuorumKey => VersionRange { min: 0, max: 1 },
+            ApiKey::AlterPartitionKey => VersionRange { min: 0, max: 3 },
+            ApiKey::UpdateFeaturesKey => VersionRange { min: 0, max: 1 },
+            ApiKey::EnvelopeKey => VersionRange { min: 0, max: 0 },
+            ApiKey::FetchSnapshotKey => VersionRange { min: 0, max: 0 },
+            ApiKey::DescribeClusterKey => VersionRange { min: 0, max: 1 },
+            ApiKey::DescribeProducersKey => VersionRange { min: 0, max: 0 },
+            ApiKey::BrokerRegistrationKey => VersionRange { min: 0, max: 3 },
+            ApiKey::BrokerHeartbeatKey => VersionRange { min: 0, max: 1 },
+            ApiKey::UnregisterBrokerKey => VersionRange { min: 0, max: 0 },
+            ApiKey::DescribeTransactionsKey => VersionRange { min: 0, max: 0 },
+            ApiKey::ListTransactionsKey => VersionRange { min: 0, max: 1 },
+            ApiKey::AllocateProducerIdsKey => VersionRange { min: 0, max: 0 },
+            ApiKey::ConsumerGroupHeartbeatKey => VersionRange { min: 0, max: 0 },
+            ApiKey::ConsumerGroupDescribeKey => VersionRange { min: 0, max: 0 },
+            ApiKey::ControllerRegistrationKey => VersionRange { min: 0, max: 0 },
+            ApiKey::GetTelemetrySubscriptionsKey => VersionRange { min: 0, max: 0 },
+            ApiKey::PushTelemetryKey => VersionRange { min: 0, max: 0 },
+            ApiKey::AssignReplicasToDirsKey => VersionRange { min: 0, max: 0 },
+            ApiKey::ListClientMetricsResourcesKey => VersionRange { min: 0, max: 0 },
+            ApiKey::DescribeTopicPartitionsKey => VersionRange { min: 0, max: 0 },
+        }
+    }
+
+    /// Iterate through every ApiKey variant in the order of the internal code.
+    pub fn iterate_all() -> impl Iterator<Item = ApiKey> {
```

Comment:
Nit: Any reason not just to call this `iter`? It seems like that's what [strum](https://docs.rs/strum_macros/latest/strum_macros/derive.EnumIter.html) does so is potentially idiomatic for enum variants.

Reply:
yeah, good point, I'll rename it